site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+
+```typescript title="src/commands/mod.ts"
+import { ApplicationCommandOption, ApplicationCommandTypes, Bot, Collection, Interaction } from "../../deps.ts";
+
+// Commands
+import { ping } from "./cmd/ping.ts";
+
+// Subcommands are just commands that can't define other subcommands.
+export type subCommand = Omit<Command, "subcommands">;
+
+export type subCommandGroup = {
+  name: string;
+  subCommands: subCommand[];
+};
+
+export interface Command {
+  name: string;
+  description: string;
+  usage?: string[];
+  options?: ApplicationCommandOption[];
+  type: ApplicationCommandTypes;
+  /** Defaults to `Guild` */
+  scope?: "Global" | "Guild";
+  execute: (bot: Bot, interaction: Interaction) => unknown;
+  subcommands?: Array<subCommandGroup | subCommand>;
+}
+
+export const commands = new Map<string, Command>();
+
+export function addCommand(command: Command) {
+  commands.set(command.name, command);
+  return command;
+};
+
+addCommand(ping);
+```
+
+You may have also noticed a helper function to add a command, this just makes sure that the command's name in the map
+matches it's name in the command map. Anyways, you may have also noticed the definitions of `subCommand` and
+`subCommandGroup`; these allow us to define the structure of sub-commands as well (because discord supports them, so we
+should too).
+
+## Defining Commands
+
+All of our commands will follow the same structure:
+
+```typescript title="src/commands/cmd/ping.ts"
+import { addCommand } from "../mod.ts";
+import { ApplicationCommandTypes } from "../../../deps.ts";
+import { snowflakeToTimestamp, humanizeMilliseconds } from "../../helpers.ts"
+
+export const ping = addCommand({
+  // this command will be used `/ping` in discord.
+  name: "ping",
+  // this will show up as the user is typing the command.
+  description: "Ping the bot.",
+  // this tells discord that we're going to be receiving text input, this is still required even if we aren't using it.
+  type: ApplicationCommandTypes.ChatInput,
+  // we should be able to DM the bot to ask for it's ping.
+  scope: "Global",
+  // this runs when the bot receives the interaction corresponding to this command.
+  execute: async (bot, interaction) => {
+    // Use the helper function to get the time when the command was executed.
+    const speed = Date.now() - snowflakeToTimestamp(interaction.id);
+    // Send a reply to the user with the time to receive the interaction.
+    await bot.helpers.sendInteractionResponse(
+      interaction.id,
+      interaction.token,
+      {
+        type: InteractionResponseTypes.ChannelMessageWithSource,
+        data: {
+          content: `🏓 Pong! Ping ${speed}ms (${humanizeMilliseconds(speed)})`,
+        },
+      },
+    );
+  }
+});
+```
+
+We should also add the following helper functions to `src/utils/helpers.ts`:
+
+```typescript title="src/utils/helpers.ts"
+/** Turn a discord snowflake into a millisecond value. */
+export function snowflakeToTimestamp(id: bigint) {
```

Comment:
This should be something in the helper plugin, id recommend using this chance to encourage using that